packages/flutter_tools/gradle/module_plugin_loader.gradle

```diff
@@ -35,6 +37,18 @@ if (pluginsFile.exists()) {
 
 gradle.getGradle().projectsLoaded { g ->
     g.rootProject.beforeEvaluate { p ->
+        p.subprojects { sp ->
+            String sanitizedPluginName = sp.name.replace(":", "");
```

Comment:
What do the input names look like here? This transformation looks like it could create collisions between plugins.

Reply:
I changed the input name to `subproject`. Maybe It is better here.

Reply:
Changing the variable name doesn't address my question, which was about the values. What do the values here look like?

This transformation would cause "a:bc" and "ab:c" to have that same output directory, which would break things. I'm trying to understand what the possible inputs are to know if that's a problem.

Reply:
The reason why the transformation was ben done here is that the subproject's name always looks like `:abcde` or `ab:c` , the name `abcde` should contained in `object.plugins.android.name` and the name `abc` will not, after we removed the `:`.

`object.plugins.android.name` was generated by `.flutter-plugins-dependencies`, It contains all the flutter plugin's native dependencies which will inject to host project.

Reply:
So ,even if the host project contain a dependency named "ab:c", `object.plugins.android.name.contains(pluginName)` will exclude it.

Reply:
> The reason why the transformation was ben done here is that the subproject's name always looks like `:abcde` or `ab:c`

What are actual examples of plugin subproject names? Are they like the first example you gave, or the second?

> the name `abcde` should contained in `object.plugins.android.name` and the name `abc` will not, after we removed the `:`.

If `abcde` is in the set, then (per my comment above) both `:abcde` and `ab:cde` would pass, and would attempt to use the same build directory, which would be bad. Is there something I'm missing that would prevent that from happening? If not, that's a bug in the logic here.

Reply:
I agree that this string transformation here is likely to cause naming conflicts. It seems like it would be straightforward for two similarly named packages where one name was a substring of another or have different colon positioning to both pass this check and conflict.

Reply:
Would it be simpler to just treat the colon as a directory separator and have the files in a subdirectory, for example `ab:c` would map to `<blah>/ab/c`. Since the goal is to create unique separate build directories, removing identifying characters like ':' seems to be counterproductive.

Reply:
With the latest master It seems that there's no `:` before project's name.
I removed It and the test file also returned 0.

Reply:
@stuartmorgan Re-request review manually.